src/domains/ecommerce-app/api/ecommerce-checkout/v1/_auth_request.xml.tpl

```diff
@@ -18,26 +18,47 @@
         }
         return "";
         }"/>
+        <!-- START set pgsId -->
         <set-variable name="XPAYPspsList" value="${ecommerce_xpay_psps_list}"/>
         <set-variable name="VPOSPspsList" value="${ecommerce_vpos_psps_list}"/>
         <set-variable name="NPGPspsList" value="${ecommerce_npg_psps_list}"/>
-        <set-variable name="pspId" value="@(((string)((JObject)context.Request.Body.As<JObject>(preserveContent: true))["pspId"]))"/>
+        <set-variable name="requestBody" value="@((JObject)context.Request.Body.As<JObject>(true))" />
+
+        <set-variable name="pspId" value="@(((string)((JObject)context.Variables["requestBody"])["pspId"]))"/>
+        <set-variable name="detailType" value="@((string)((JObject)((JObject)context.Variables["requestBody"])["details"])["detailType"])"/>
+
         <set-variable name="pgsId" value="@{
-        string[] xpayList = ((string)context.Variables["XPAYPspsList"]).Split(',');
-        string[] vposList = ((string)context.Variables["VPOSPspsList"]).Split(',');
-        string[] npgList = ((string)context.Variables["NPGPspsList"]).Split(',');
-        string pspId = (string)(context.Variables.GetValueOrDefault("pspId",""));
-        if (xpayList.Contains(pspId)) {
-            return "XPAY";
-        }
-        if (vposList.Contains(pspId)) {
-            return "VPOS";
-        }
-        if (npgList.Contains(pspId)) {
-            return "NPG";
-        }
-        return "";
-    }"/>
+            
+            string[] xpayList = ((string)context.Variables["XPAYPspsList"]).Split(',');
+            string[] vposList = ((string)context.Variables["VPOSPspsList"]).Split(',');
+            string[] npgList = ((string)context.Variables["NPGPspsList"]).Split(',');
+            
+            string pspId = (string)(context.Variables.GetValueOrDefault("pspId",""));
+            string detailType = (string)(context.Variables.GetValueOrDefault("detailType",""));
+            
+            string pgsId = "";
+
+            // card -> ecommerce with PGS request
+            
+            if ( detailType == "card" ){                                    
+
+                if (xpayList.Contains(pspId)) {
+
+                    pgsId = "XPAY";
+                } else if (vposList.Contains(pspId)) {
+
+                    pgsId = "VPOS";
+                }
+
+            // cards -> ecommerce with NPG request
+
+            } else if ( detailType == "cards" && npgList.Contains(pspId)){      
```

Comment:
As for now we have also `wallet`, `apm` and `redirect` detailTypes, the first two used for onboarded/guest payments, the last for redirect payment flows.

Since the gateways will be 3 as for now (two in the future when pgs will be shutdown) what about changing this else if condition to check against `redirect` detailType (and set pgsId to redirect) and then in the else branch assume that it's an NPG transaction?

Maybe we can also add a psp id based filtering for redirect transactions, what do you think?

Reply:
redirect added :-) thanks!